### PR TITLE
Show that permuting a list index forwards then backwards is the identity

### PR DESCRIPTION
**Pull request by Twisol**
This PR adds the following property to `Data.List.Relation.Binary.Permutation.Propositional.Properties`, which I had need of in some of my own developments. I asked about it [on Zulip](https://agda.zulipchat.com/#narrow/stream/259644-newcomers/topic/Lemmas.20for.20permutations.20on.20indices.20in.20the.20stdlib.3F/near/277941007), and @MatthewDaggitt indicated it might be welcome in the stdlib.

```agda
∈-resp-↭[sym-σ∘σ] : {xs ys : List A} {x : A} →
                    (σ : xs ↭ ys) →
                    (ix : x ∈ xs) →
                    ∈-resp-↭ (trans σ (↭-sym σ)) ix ≡ ix
```

To be honest, I have no idea what this ought to be called, or where in the file it should live. I'm very happy to call it whatever you all think makes the most sense 😅 It was originally called `σ-inv` in my developments, but I suspect that's far too ambiguous for the stdlib.

### CODE REVIEW

**Comment by Twisol** [comment]
Yes, I was hunting for that property in the stdlib before tried to prove this result, but I couldn't seem to find it. Does that exist somewhere already? Or if not, would you prefer a proof of that property instead of this one?

**Comment by Twisol** [comment]
(As I explore that, I notice that `trans σ (↭-sym σ)` does not normalize to `refl`, because `trans` is merely a constructor. It should work for the smarter `↭-trans` though.)

**Comment by Twisol** [comment]
Is there an existing non-definitional equivalence for permutations? I can't find one; and even `↭-trans σ (↭-sym σ)` is simply constructed as `trans σ (↭-sym σ)` for nontrivial `σ`.

Put somewhat differently, I can't find anything that computes a `trans`-free composition of permutations, which I think would be necessary to prove this by appeal to an identity permutation.

**Comment by MatthewDaggitt** [comment]
So, no we have no definition for equality of permutations (although we do have one for `Data.Fin.Permutation`). The addition of equality or a function that computes a trans free representation would also be welcome, but that's a fair bit of work so it's up to @Twisol whether to add it or not?

File: src/Data/List/Relation/Binary/Permutation/Propositional/Properties.agda [hunk 1]
```diff
@@ -87,6 +87,25 @@ Any-resp-↭ (trans p p₁) wit                 = Any-resp-↭ p₁ (Any-resp-
 ∈-resp-↭ : ∀ {x : A} → (x ∈_) Respects _↭_
 ∈-resp-↭ = Any-resp-↭
 
+∈-resp-↭[sym-σ∘σ] : {xs ys : List A} {x : A} →
```
[the remark this answers is not in the document]
Comment (Twisol) [reply src/Data/List/Relation/Binary/Permutation/Propositional/Properties.agda:90]: I like it!

**Comment by Twisol** [comment]
> or a function that computes a trans free representation

If I understand how permutations have been defined here, I think a permutation without any `trans` constructors could only displace any element one step to the left or right. So I think an alternate representation would be necessary -- and that's rather more invasive than I was hoping to get 😆

File: CHANGELOG.md [hunk 2]
```diff
@@ -1163,6 +1163,14 @@ Other minor changes
   ∷ʳ-++ : xs ∷ʳ a ++ ys ≡ xs ++ a ∷ ys
   ```
 
+* Added new proofs to `Data.List.Relation.Binary.Permutation.Propositional.Properties`:
+  ```agda
+  ∈-resp-↭[sym-σ∘σ] : {xs ys : List A} {x : A} →
```
[the remark this answers is not in the document]
Comment (Twisol) [reply CHANGELOG.md]: Ah, understood! I've made this change (and sorted this addition alphabetically under some other `Data.List.Relation` changes I'd overlooked.)

**Comment by Twisol** [comment]
I realized that this proof works just as well for any `Any P`, not just for membership, so I generalized it to `Any-resp-[σ⁻¹∘σ]` and reduced `∈-resp-[σ⁻¹∘σ]` to a thin wrapper around it, much like the definition of `∈-resp-↭` above it.

**Comment by MatthewDaggitt** [comment]
>  So I think an alternate representation would be necessary -- and that's rather more invasive than I was hoping to get

That's fair enough :smile: 

Looks good to me. I'll leave it up for a few more days in case anyone else has any comments, and then I'll merge it in.

(Just a heads-up that we tend to avoid force-pushing as it can mess up other people who have also cloned the branch to play around with it. When we merge the changes we also squash, so the messy commit history gets erased then.)

**Comment by Twisol** [comment]
> Looks good to me. I'll leave it up for a few more days in case anyone else has any comments, and then I'll merge it in.
> 
> (Just a heads-up that we tend to avoid force-pushing as it can mess up other people who have also cloned the branch to play around with it. When we merge the changes we also squash, so the messy commit history gets erased then.)

Ah, fair! I'm used to a rebase-oriented style where the branch history is kept around (so it's important that it be "nice"). I'll keep that in mind for this repository 😄

**Comment by MatthewDaggitt** [comment]
Merging in. Thanks again for the PR!

**Pull request merged**